test/performance/specs/site-editor.spec.js

```diff
@@ -206,6 +207,83 @@ test.describe( 'Site Editor Performance', () => {
 			} );
 		}
 	} );
+
+	test.describe( 'Loading Patterns', () => {
+		test.beforeAll( async ( { requestUtils } ) => {
+			await requestUtils.activateTheme( 'twentytwentyfour' );
+		} );
+
+		test.afterAll( async ( { requestUtils } ) => {
+			await requestUtils.activateTheme( 'twentytwentyfour' );
+		} );
+
+		test( 'Run the test', async ( { page, admin, perfUtils, editor } ) => {
+			const samples = 10;
+			for ( let i = 1; i <= samples; i++ ) {
+				// We want to start from a fresh state each time, without
+				// queries or patterns already cached.
+				await admin.visitSiteEditor( {
+					postId: 'twentytwentyfour//home',
+					postType: 'wp_template',
+					canvas: 'edit',
+				} );
+				await editor.openDocumentSettingsSidebar();
+				await page
+					.getByRole( 'button', {
+						name: 'Actions',
+					} )
+					.click();
+
+				// Wait for the browser to be idle before starting the monitoring.
+				// eslint-disable-next-line no-restricted-syntax
+				await page.waitForTimeout( BROWSER_IDLE_WAIT );
+
+				const startTime = performance.now();
+
+				await page
+					.getByRole( 'menuitem', { name: 'Replace template' } )
+					.click();
+
+				const patterns = [
+					'Blogging home template',
+					'Business home template',
+					'Portfolio home template with post featured images',
+					'Blogging index template',
+				];
+
+				await Promise.all(
+					patterns.map( async ( pattern ) => {
+						const canvas = await perfUtils.getCanvas(
+							page
+								.getByRole( 'option', {
+									name: pattern,
+									exact: true,
+								} )
+								.getByTitle( 'Editor canvas' )
+						);
+
+						// Wait until the first block is rendered AND all
+						// patterns are replaced.
+						await Promise.all( [
+							canvas.locator( '.wp-block' ).first().waitFor(),
+							page.waitForFunction(
+								() =>
+									document.querySelectorAll(
+										'[data-type="core/pattern"]'
+									).length === 0
+							),
+						] );
+					} )
+				);
+
+				const endTime = performance.now();
```

Comment:
I think relying on waiting and timers like that might not be very stable but we can see over time.

Reply:
It's what we're using for the other patterns load test as well, and also load in general I think